sdk/common/src/main/java/io/opentelemetry/sdk/resources/Resource.java

```diff
@@ -16,19 +16,34 @@
 import io.opentelemetry.api.common.AttributesBuilder;
 import io.opentelemetry.api.internal.StringUtils;
 import io.opentelemetry.api.internal.Utils;
+import java.util.Arrays;
 import java.util.Objects;
 import java.util.Properties;
 import java.util.ServiceLoader;
+import java.util.Set;
+import java.util.stream.Collectors;
 import javax.annotation.Nullable;
 import javax.annotation.concurrent.Immutable;
 
 /**
  * {@link Resource} represents a resource, which capture identifying information about the entities
  * for which signals (stats or traces) are reported.
+ *
+ * <p>To disable any {@link ResourceProvider} found on the classpath from being recognized, set the
+ * fully qualified class names of the {@link ResourceProvider} implementations as a comma separated
+ * list to the system property {@code -Dotel.java.disabled.resource_providers} or the {@code
+ * OTEL_JAVA_DISABLED_RESOURCE_PROVIDERS} environment variable.
  */
 @Immutable
 @AutoValue
 public abstract class Resource {
+
+  private static final String OTEL_JAVA_DISABLED_RESOURCES_PROVIDERS_PROPERTY_KEY =
+      "otel.java.disabled.resource_providers";
```

Comment:
instrumentation repo uses `-` instead of `_` in property names, e.g. `otel.java.disabled.resource-providers`, following spring convention: https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/1414#issuecomment-733181398

Reply:
this isn't a new property. If we want to change it, that should be a separate thing to track.

Reply:
👍 I'll open an issue for consideration